publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -0,0 +1,175 @@
+
+    public ResourceService(AmazonDynamoDB client, Clock clock) {
+        tableName = RESOURCES_TABLE_NAME;
+        this.client = client;
+        this.clockForTimestamps = clock;
+    }
+
+    public ResourceService() {
+        this(AmazonDynamoDBClientBuilder.defaultClient(), Clock.systemDefaultZone());
+    }
+
+    public void createResource(Resource resource) throws ConflictException {
+        resource.setCreatedDate(clockForTimestamps.instant());
+        TransactWriteItem[] transactionItems = transactionItemsForNewResourceInsertion(resource);
+        TransactWriteItemsRequest putRequest = newTransactWriteItemsRequest(transactionItems);
+        sendRequest(putRequest);
+    }
+
+    public void updateResource(Resource resourceUpdate)  {
+        ResourceDao resourceDao = new ResourceDao(resourceUpdate);
+        String conditionExpression = primaryKeyHasNotChangedCondition();
+
+        Map<String, AttributeValue> valuesMap = primaryKeyAttributeValuesMap(resourceDao);
+
+        PutItemRequest putItemRequest = new PutItemRequest()
+            .withItem(toDynamoFormat(resourceDao))
+            .withTableName(tableName)
+            .withConditionExpression(conditionExpression)
+            .withExpressionAttributeNames(PRIMARY_KEY_PLACEHOLDERS_AND_ATTRIBUTE_NAMES_MAPPING)
+            .withExpressionAttributeValues(valuesMap);
+
+        client.putItem(putItemRequest);
+    }
+
+    public Resource getResource(UserInstance userInstance, SortableIdentifier resourceIdentifier)
+        throws NotFoundException {
+        return getResource(createQueryObject(userInstance, resourceIdentifier.toString()));
+    }
+
+    public Resource getResource(UserInstance userInstance, String resourceIdentifier) throws NotFoundException {
+        return getResource(createQueryObject(userInstance, resourceIdentifier));
+    }
+
+    public Resource getResource(Resource resource) throws NotFoundException {
+        Map<String, AttributeValue> primaryKey = new ResourceDao(resource).primaryKey();
+        GetItemResult getResult = getResourceByPrimaryKey(primaryKey);
+        ResourceDao fetchedDao = parseResult(getResult);
+        return fetchedDao.getResource();
+    }
+
+    public void updateOwner(String identifier, UserInstance oldOwner, UserInstance newOwner) throws NotFoundException {
+        Resource existingResource = getResource(oldOwner, identifier);
+        Resource newResource = updateResourceOwner(newOwner, existingResource);
+        TransactWriteItem deleteAction = newDeleteTransactionItem(existingResource);
+        TransactWriteItem insertionAction = newPutTransactionItem(createNewTransactionPutDataEntry(newResource));
+        TransactWriteItemsRequest request = newTransactWriteItemsRequest(deleteAction, insertionAction);
+        client.transactWriteItems(request);
+    }
+
+    private Map<String, AttributeValue> primaryKeyAttributeValuesMap(ResourceDao resourceDao) {
+        return Map.of(PARTITION_KEY_VALUE_PLACEHOLDER,
```

Comment:
Probably put the placeholders directly into the substitution template as this makes the template more readable, then extract the substitution template to a named constant